plugins/store/kics/main.go

```diff
@@ -0,0 +1,272 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/Checkmarx/kics/pkg/model"
+	"github.com/Checkmarx/kics/pkg/printer"
+	"github.com/Checkmarx/kics/pkg/progress"
+	"github.com/Checkmarx/kics/pkg/scan"
+	"github.com/hashicorp/hcl/v2/hclsimple"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+
+	"github.com/openclarity/vmclarity/plugins/sdk/cmd/run"
+	"github.com/openclarity/vmclarity/plugins/sdk/types"
+)
+
+var mapKICSSeverity = map[model.Severity]types.MisconfigurationSeverity{
+	model.SeverityHigh:   types.MisconfigurationSeverityHigh,
+	model.SeverityMedium: types.MisconfigurationSeverityMedium,
+	model.SeverityLow:    types.MisconfigurationSeverityLow,
+	model.SeverityInfo:   types.MisconfigurationSeverityInfo,
+	model.SeverityTrace:  types.MisconfigurationSeverityInfo,
+}
+
+//nolint:containedctx
+type KICSScanner struct {
+	healthz bool
+	status  *types.Status
+	cancel  context.CancelFunc
+}
+
+type ScanParametersConfig struct {
+	PreviewLines     int      `json:"preview-lines" yaml:"preview-lines" toml:"preview-lines" hcl:"preview-lines"`
+	Platform         []string `json:"platform" yaml:"platform" toml:"platform" hcl:"platform"`
+	MaxFileSizeFlag  int      `json:"max-file-size-flag" yaml:"max-file-size-flag" toml:"max-file-size-flag" hcl:"max-file-size-flag"`
+	DisableSecrets   bool     `json:"disable-secrets" yaml:"disable-secrets" toml:"disable-secrets" hcl:"disable-secrets"`
+	QueryExecTimeout int      `json:"query-exec-timeout" yaml:"query-exec-timeout" toml:"query-exec-timeout" hcl:"query-exec-timeout"`
+	Silent           bool     `json:"silent" yaml:"silent" toml:"silent" hcl:"silent"`
+	Minimal          bool     `json:"minimal" yaml:"minimal" toml:"minimal" hcl:"minimal"`
+}
+
+func (s *KICSScanner) Healthz() bool {
+	return s.healthz
+}
+
+func (s *KICSScanner) Metadata() *types.Metadata {
+	return &types.Metadata{
+		ApiVersion: types.Ptr(types.APIVersion),
+		Name:       types.Ptr("KICS"),
+		Version:    types.Ptr("v1.7.13"),
+	}
+}
+
+func (s *KICSScanner) Start(config *types.Config) {
+	log.Infof("Starting scanner with config: %+v\n", config)
+
+	go func() {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Duration(config.TimeoutSeconds)*time.Second)
+		s.cancel = cancel
+		defer cancel()
+
+		log.Infof("Scanner is running...")
+		s.SetStatus(types.NewScannerStatus(types.Running, types.Ptr("Scanner is running...")))
+		tmp := os.TempDir()
+
+		clientConfig, err := s.createScanParametersConfig(config.File)
+		if err != nil {
+			log.Errorf("Failed to parse config file: %v", err)
+			s.SetStatus(types.NewScannerStatus(types.Failed, types.Ptr(fmt.Errorf("failed to parse config file: %w", err).Error())))
+			return
+		}
+
+		outputFile := filepath.Join(tmp, "kics.json")
```

Comment:
maybe `rawOutputFile` here

Reply:
I would also remove `tmp` variable. Set `rawOutputFile := filepath.Join(os.TempDir(), "kics.json")` then, use `filepath.Dir(rawOutputFile)` for the file directory

Reply:
```go
		tmp := os.TempDir()

		clientConfig, err := s.createScanParametersConfig(config.File)
		if err != nil {
			slog.Error("Failed to parse config file", slog.Any("error", err))
			s.SetStatus(types.NewScannerStatus(types.Failed, types.Ptr(fmt.Errorf("failed to parse config file: %w", err).Error())))
			return
		}

		rawOutputFile := filepath.Join(tmp, "kics.json") // tmp path used here
		c, err := scan.NewClient(
			&scan.Parameters{
				Path:             []string{config.InputDir},
				QueriesPath:      []string{"../../../queries"},
				PreviewLines:     clientConfig.PreviewLines,
				Platform:         clientConfig.Platform,
				OutputPath:       tmp, // tmp path used here
				MaxFileSizeFlag:  clientConfig.MaxFileSizeFlag,
				DisableSecrets:   clientConfig.DisableSecrets,
				QueryExecTimeout: clientConfig.QueryExecTimeout,
				OutputName:       "kics",
			},
			&progress.PbBuilder{Silent: clientConfig.Silent},
			printer.NewPrinter(clientConfig.Minimal), //nolint:forbidigo
```

As we discovered KICS constructs the output using `OutputPath` and `outputName`, so we need to save it and pass it to the scan parameters.

But we can do something like this:

```go
		rawOutputFile := filepath.Join(os.TempDir(), "kics.json")

		c, err := scan.NewClient(
			&scan.Parameters{
				Path:             []string{config.InputDir},
				QueriesPath:      []string{"../../../queries"},
				PreviewLines:     clientConfig.PreviewLines,
				Platform:         clientConfig.Platform,
				OutputPath:       filepath.Dir(rawOutputFile),
				MaxFileSizeFlag:  clientConfig.MaxFileSizeFlag,
				DisableSecrets:   clientConfig.DisableSecrets,
				QueryExecTimeout: clientConfig.QueryExecTimeout,
				OutputName:       "kics",
			},
			&progress.PbBuilder{Silent: clientConfig.Silent},
			printer.NewPrinter(clientConfig.Minimal), //nolint:forbidigo
		)
```

I kinda like the last approach better.